source/modulo_components/test/python/conftest.py

```diff
@@ -1,9 +1,95 @@
+from concurrent.futures import Future
+
+import clproto
 import pytest
 import rclpy
+import state_representation as sr
+from lifecycle_msgs.msg import Transition
+from lifecycle_msgs.srv import ChangeState
+from modulo_core import EncodedState
+from rclpy.node import Node
+from rclpy.parameter import Parameter
+
+
+@pytest.fixture
+def random_state():
+    return sr.CartesianPose.Random("test")
 
 
 @pytest.fixture
 def ros_context():
     rclpy.init()
     yield
-    rclpy.shutdown()
\ No newline at end of file
+    rclpy.shutdown()
+
+
+@pytest.fixture
+def ros_exec():
+    rclpy.init()
+    executor = rclpy.executors.SingleThreadedExecutor()
+    yield executor
+    rclpy.shutdown()
+
+
+@pytest.fixture
+def minimal_cartesian_output(request, random_state):
+    def _make_minimal_cartesian_output(component_type, topic):
+        component = component_type("minimal_cartesian_output", parameter_overrides=[Parameter("period", value=0.1)])
```

Comment:
it's unrelated to this PR but I've had this small backlog issue noted down for a while... the default component period in Python is already 0.1 (while in C++ it's 1.0). To be honest I'd prefer to revert the C++ default back to 0.1. I think it's a good balance between being responsive for the user while not being too CPU intensive, and just because 1.0 is a more "pure unit number" I don't think it's a good default period.
```suggestion
        component = component_type("minimal_cartesian_output")
```